### PR TITLE
Selector

### PR DESCRIPTION
**Pull request by zshaheen**
* Added `selectors` parameter, w/ tests.

### CODE REVIEW

File: cdp/test/test_cdp_parser.py [hunk 1]
```diff
@@ -385,27 +385,31 @@ def test_get_parameters(self):
...
             self.assertEqual(p.vars, ['v1'])
 
             self.cdp_parser.add_args_and_values(['-d', 'test_get_parameters.cfg', '-v', 'v3'])
             p = self.cdp_parser.get_parameters(default_vars=False, cmd_default_vars=False, argparse_vals_only=False)[0]
             self.assertEqual(p.vars, ['v3'])
+
             self.cdp_parser.add_args_and_values(['-d', 'test_get_parameters.cfg', '-v', 'v3', '--selectors', 'vars'])
-            p = self.cdp_parser.get_parameters(default_vars=False, cmd_default_vars=False, argparse_vals_only=False)[0]
-            self.assertEqual(p.vars, ['v2'])
+            p = self.cdp_parser.get_parameters(default_vars=False, cmd_default_vars=False, argparse_vals_only=False)
```
Comment (doutriaux1) [inline cdp/test/test_cdp_parser.py]: why do look at p == [] when it used to be p.vars == ['v2']

Comment (zshaheen) [reply cdp/test/test_cdp_parser.py]: In master, I made a change to line 404, which added the `selectors` argument.
```
self.cdp_parser.add_args_and_values(['-d', 'test_get_parameters.cfg', '-v', 'v3', '--selectors', 'vars'])
```
This is why it's not red. Should also be marked as red, so I'm not cheating.
Forgive that mistake.

**Comment by doutriaux1** [comment]
@zshaheen fix conflict and merge whenever you want

**Comment by zshaheen** [comment]
@doutriaux1 Thanks.

**Pull request merged**